src/modules/common/containers/Helmet.js

```diff
@@ -3,36 +3,70 @@
 import * as React from 'react'
 import type { StateType } from '../../app/StateType'
 import { connect } from 'react-redux'
-import { CategoriesMapModel, EventModel, LanguageModel, PoiModel, CityModel } from '@integreat-app/integreat-api-client'
+import {
+  CategoriesMapModel,
+  EventModel,
+  LanguageModel,
+  PoiModel,
+  CityModel,
+  ExtraModel,
+  WohnenOfferModel
+} from '@integreat-app/integreat-api-client'
 import ReactHelmet from 'react-helmet'
 
 import type { Location } from 'redux-first-router'
-import getLanguageChangePath from '../../app/getLanguageChangePath'
+import { getRouteConfig } from '../../app/route-configs/index'
+import type { TFunction } from 'react-i18next'
+import compose from 'lodash/fp/compose'
+import { withNamespaces } from 'react-i18next'
+import CategoriesRouteConfig from '../../app/route-configs/CategoriesRouteConfig'
+import { LANDING_ROUTE } from '../../app/route-configs/LandingRouteConfig'
+import type { GetPageTitleParamsType } from '../../app/route-configs/RouteConfigInterface'
 
 type PropsType = {|
-  title: string,
-  categories: CategoriesMapModel,
-  events: Array<EventModel>,
-  pois: Array<PoiModel>,
-  cities: Array<CityModel>,
-  languages: Array<LanguageModel>,
+  getPageTitle: GetPageTitleParamsType => string,
+  categories: ?CategoriesMapModel,
+  events: ?Array<EventModel>,
+  pois: ?Array<PoiModel>,
+  cities: ?Array<CityModel>,
+  languages: ?Array<LanguageModel>,
   location: Location,
-  metaDescription?: string
+  extras: ?Array<ExtraModel>,
+  offers: ?Array<WohnenOfferModel>,
+  t: TFunction
 |}
 
 export class Helmet extends React.Component<PropsType> {
   getLanguageLinks (): React.Node {
     const {languages, events, pois, categories, location} = this.props
-    return languages && languages
-      .map(language => {
-        const path = getLanguageChangePath({events, pois, categories, languageCode: language.code, location})
-        return <link key={language.code} rel='alternate' hrefLang={language.code} href={path} />
-      })
+    if (!languages) {
+      return null
+    }
+    return languages.map(language => {
+      const path = getRouteConfig(location.type).getLanguageChangePath(
+        {events, pois, categories, location, language: language.code})
+      return path && <link key={language.code} rel='alternate' hrefLang={language.code} href={path} />
+    })
+  }
+
+  getMetaDescription (): ?string {
```

Comment:
Make this a per-route-configurable function (add to interface, evaluate in Switcher and make it a ?string prop in Helmet)